Added composite index to posts table for type,status

9larsons opened the pull request:
ref https://linear.app/tryghost/issue/CFR-35
- performance improvement intended for the content api/get helpers

The posts table is shared by posts and pages and seldom is queried for both. It makes sense to add an index on type, and from the perspective of the content API, also on status as you're almost only ever querying for published posts or published pages.

9larsons committed:
Added composite index to posts table for type,status

ref https://linear.app/tryghost/issue/CFR-35
- performance improvement intended for the content api/get helpers

The posts table is shared by posts and pages and seldom is queried for both. It makes sense to add an index on type, and from the perspective of the content API, also on status as you're almost only ever querying for published posts or published pages.

9larsons committed:
update unit tests

9larsons commented:
Index is added in a few ms for a test set of 20k posts. This may be a bit longer for the couple biggest customers - I think the largest has ~125k posts, but all in all is very quick.

daniellockyer commented on ghost/core/core/server/data/migrations/versions/5.86/2024-06-20-17-02-15-add-posts-type-status-index.js:
Version needs bumping

9larsons committed:
Updated migration minor version

we had a release while this PR was sitting so we need to make sure this runs on the next released version
